src/basic/fileio.c

```diff
@@ -264,26 +264,29 @@ int verify_file(const char *fn, const char *blob, bool accept_extra_nl) {
         return 1;
 }
 
-int read_full_stream(
+int read_full_stream_full(
                 FILE *f,
+                bool secure,
+                decoder_t decoder,
```

Comment:
hmm, maybe intrdouce a flags enum "ReadFullFileFlags" or so, with one field READ_FULL_FILE_SECURE, and maybe another one READ_FILL_FILE_UNHEX or so? i.e. I have the suspicion we are not going to have that many more decoders (maybe base64), but that's it hence a flag for that should suffice and would certainly help keeping our API smaller?